src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_black_knight.cpp

```diff
@@ -372,7 +571,177 @@ class npc_black_knight_skeletal_gryphon : public CreatureScript
+            GetTarget()->RemoveAura(SPELL_DEATH_RESPITE_DND);
+            GetTarget()->CastSpell(GetTarget(), SPELL_FEIGN_DEATH, true);
+            GetTarget()->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE);
+        }
+
+        void Register() override
+        {
+            AfterEffectRemove += AuraEffectRemoveFn(spell_black_knight_deaths_push_AuraScript::HandleScript, EFFECT_1, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_black_knight_deaths_push_AuraScript();
+    }
+};
+
+class spell_black_knight_obliterate : public SpellScriptLoader
+{
+public:
+    spell_black_knight_obliterate() : SpellScriptLoader("spell_black_knight_obliterate") { }
+
+    class spell_black_knight_obliterate_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_black_knight_obliterate_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_BLOOD_PLAGUE) ||
+                !sSpellMgr->GetSpellInfo(SPELL_FROST_FEVER) ||
+                !sSpellMgr->GetSpellInfo(SPELL_BLOOD_PLAGUE_H) ||
+                !sSpellMgr->GetSpellInfo(SPELL_FROST_FEVER_H))
+                return false;
+            return true;
+        }
+
+        void CalculateDamage()
+        {
+            if (!GetHitUnit())
+                return;
+            uint32 bloodPlague = GetCaster()->GetMap()->IsHeroic() ? SPELL_BLOOD_PLAGUE_H : SPELL_BLOOD_PLAGUE;
+            uint32 frostFever = GetCaster()->GetMap()->IsHeroic() ? SPELL_FROST_FEVER_H : SPELL_FROST_FEVER;
+
+            int32 damage = GetHitDamage();
+            int32 bonus = 0;
+            if (GetHitUnit()->HasAura(frostFever))
+            {
+                bonus += int32(damage * 0.3f);
+                GetHitUnit()->RemoveAurasDueToSpell(frostFever);
+            }
+            if (GetHitUnit()->HasAura(bloodPlague))
+            {
+                bonus += int32(damage * 0.3f);
+                GetHitUnit()->RemoveAurasDueToSpell(bloodPlague);
+            }
+            SetHitDamage(damage + bonus);
+        }
+
+        void Register() override
+        {
+            OnHit += SpellHitFn(spell_black_knight_obliterate_SpellScript::CalculateDamage);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_black_knight_obliterate_SpellScript();
+    }
+};
+
+class spell_black_knight_army_of_the_dead : public SpellScriptLoader
+{
+public:
+    spell_black_knight_army_of_the_dead() : SpellScriptLoader("spell_black_knight_army_of_the_dead") { }
+
+    class spell_black_knight_army_of_the_dead_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_black_knight_army_of_the_dead_AuraScript);
+
+        void RemoveFlag(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            // that is wrong, movement disabled by spell (channel)
```

Comment:
> `// that is wrong`

^i agree with that comment, this should not be necessary if you fix the `_ROOT` hack on spell cast too